Content.Shared/_EE/Items/OniOnlySystem.cs

```diff
@@ -0,0 +1,52 @@
+using Content.Server.Abilities.Oni;
+using Content.Shared.Ghost;
+using Content.Shared.Hands.EntitySystems;
+using Content.Shared.Popups;
+using Content.Shared.StatusEffect;
+using Content.Shared.Stunnable;
+using Content.Shared.Weapons.Melee.Events;
+using Robust.Shared.Audio;
+using Robust.Shared.Audio.Systems;
+
+
+namespace Content.Shared._EE.Item;
+
+public sealed class OniOnlySystem : EntitySystem
+{
+    [Dependency] private readonly SharedStunSystem _stun = default!;
+    [Dependency] private readonly SharedPopupSystem _popupSystem = default!;
+    [Dependency] private readonly SharedHandsSystem _hands = default!;
+    [Dependency] private readonly StatusEffectsSystem _statusEffects = default!;
+    [Dependency] private readonly SharedAudioSystem _audioSystem = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<OniOnlyComponent, AttemptMeleeEvent>(OnMeleeAttempt);
+    }
+
+    private void OnMeleeAttempt(EntityUid uid, OniOnlyComponent component, ref AttemptMeleeEvent args)
+    {
+        bool CanUse(EntityUid? uid) => HasComp<OniComponent>(uid) || HasComp<GhostComponent>(uid);
+
+        // Specism.
+        if (CanUse(args.PlayerUid))
+            return;
+
+        args.Message = Loc.GetString("oni-only-component-attack-fail-self", ("item", uid));
```

Comment:
```suggestion
        args.Message = Loc.GetString(component.FailText, ("item", uid));
```
Make sure that the component contains a datafield for a string that matches this.